### PR TITLE
Issue-144 - ActivityType repository added, UnitOfWork injected in Ser…

### PR DESCRIPTION
**Pull request by LusineHovs**
…vice layer

### CODE REVIEW

File: src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs [hunk 1]
```diff
@@ -9,6 +10,7 @@ public class UnitOfWork : IUnitOfWork
     {
         private readonly ServiceManagerDbContext _dbContext;
         private ActivityRepository _activityRepository;
+        private ActivityTypeRepository _activityTypeRepository;
         public UnitOfWork(ServiceManagerDbContext dbContext)
         {
             _dbContext = dbContext;
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs:16]: null check here please

File: src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs [hunk 2]
```diff
@@ -24,6 +26,17 @@ public IActivityRepository Activities
             }
         }
 
+        public IActivityTypeRepository ActivityTypes
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs:29]: I would suggest naming it `ActivityTypeRepository` and to have standard `{ get; set; }`, then initialize it from ctor

File: src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs [hunk 3]
```diff
@@ -24,6 +26,17 @@ public IActivityRepository Activities
...
+                if (_activityTypeRepository == null)
+                    _activityTypeRepository = new ActivityTypeRepository(_dbContext);
+
+                return _activityTypeRepository;
+            }
+        }
+
         public void Dispose()
         {
             _dbContext.Dispose();
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs:42]: Do not dispose dbContext from here, other services in the same scope may use it

File: src/ServiceManager/Travely.ServiceManager.Service/Managers/ActivityManager.cs [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
-using System.Linq;
 using System.Threading.Tasks;
+using Travely.ServiceManager.Abstraction.Interfaces.UnitOfWorks;
 
 namespace Travely.ServiceManager.Service.Managers
 {
     public class ActivityManager : IActivityManager
     {
+        protected readonly IUnitOfWork _repos;
+        public ActivityManager(IUnitOfWork unitOfWork)
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.Service/Managers/ActivityManager.cs:11]: Please do not name them differently

Comment (martikyan) [reply src/ServiceManager/Travely.ServiceManager.Service/Managers/ActivityManager.cs:11]: Also, an empty line missing between 10, 11

File: src/ServiceManager/Travely.ServiceManager.Service/Startup.cs [hunk 5]
```diff
@@ -5,19 +5,35 @@
...
+        }
+
+        public IConfiguration Configuration { get; }
+
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddDbContext<ServiceManagerDbContext>(options =>
+            options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.Service/Startup.cs:31]: Please fix the intent here

File: src/ServiceManager/Travely.ServiceManager.Service/Startup.cs [hunk 6]
```diff
@@ -5,19 +5,35 @@
...
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddDbContext<ServiceManagerDbContext>(options =>
```
Comment (martikyan) [inline src/ServiceManager/Travely.ServiceManager.Service/Startup.cs:30]: please use `AddDbContextPool` to allow pooling dbContexts

**Review by martikyan — CHANGES_REQUESTED** [review]

File: src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs (hunk 2 above)
Comment (LusineHovs) [reply src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs:29]: ok I will change the name, regarding to initialization in ctor, don't you think that will it be better to initialize the repository when we need it?

Comment (martikyan) [reply src/ServiceManager/Travely.ServiceManager.DAL/UnitOfWorks/UnitOfWork.cs:29]: I think that readability and maintainability are a priority comparing to that kind of optimization

**Review by simonyangevorg — APPROVED** [review]

File: src/ServiceManager/Travely.ServiceManager.DAL/DbContextOptionsBuilderExtension.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Travely.ServiceManager.DAL
+{
+    public static class DbContextOptionsBuilderExtension
+    {
+        public static void UseCustomDatabaseServer(this DbContextOptionsBuilder options, IConfiguration configuration)
```
Comment (tadart) [inline src/ServiceManager/Travely.ServiceManager.DAL/DbContextOptionsBuilderExtension.cs:13]: Please use more descriptive name for a method, maybe `UseServiceManagerDatabaseServer`

File: src/ServiceManager/Travely.ServiceManager.DAL/DbContextOptionsBuilderExtension.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        {
+            options.UseSqlServer(configuration.GetConnectionString("DefaultConnection"));
+            options.UseLazyLoadingProxies();
```
Comment (tadart) [inline src/ServiceManager/Travely.ServiceManager.DAL/DbContextOptionsBuilderExtension.cs:16]: We don't need this. The navigation properties better to load explicitly by Include.

**Review by tadart — APPROVED** [review]

**Review by simonyangevorg — APPROVED** [review]

**Pull request merged**